pkg/ucp/store/client.go

```diff
@@ -18,8 +18,19 @@ package store
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"regexp"
 )
 
+// jsonPropertyPattern is the pattern for a valid JSON property name.
+var jsonPropertyPattern = "[a-zA-Z$_][a-zA-Z0-9$_]*"
```

Comment:
The "field query" feature was not documented anywhere in code and the inputs were not validated. This change documents the **current** behavior and validates that the fields are specified using the correct syntax.